openverse_catalog/dags/providers/provider_api_scripts/smk.py

```diff
@@ -1,223 +1,180 @@
-PROVIDER = prov.SMK_DEFAULT_PROVIDER
-ENDPOINT = "https://api.smk.dk/api/v1/art/search/"
-LANDING_PAGE_BASE_URL = "https://open.smk.dk/en/artwork/image/"
 IMAGE_SIZE = 2048
 
-delay_request = DelayedRequester(delay=DELAY)
-image_store = ImageStore(provider=PROVIDER)
 
-DEFAULT_QUERY_PARAMS = {
-    "keys": "*",
-    "filters": "[has_image:true],[public_domain:true]",
-    "offset": 0,
-    "rows": LIMIT,
-}
+class SmkDataIngester(ProviderDataIngester):
+    endpoint = "https://api.smk.dk/api/v1/art/search/"
+    delay = 5
+    retries = 3
+    batch_limit = 2000
+    headers = {"Accept": "application/json"}
+    providers = {"image": prov.SMK_DEFAULT_PROVIDER}
+
+    def get_media_type(self, record: dict) -> str:
+        return constants.IMAGE
+
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        if not prev_query_params:
+            return {
+                "keys": "*",
+                "filters": "[has_image:true],[public_domain:true]",
+                "offset": 0,
+                "rows": self.batch_limit,
+            }
+        return {
+            **prev_query_params,
+            "offset": prev_query_params["offset"] + self.batch_limit,
+        }
+
+    def get_batch_data(self, response_json) -> list:
+        return response_json.get("items")
+
+    @staticmethod
+    def _get_foreign_landing_url(item) -> str | None:
+        """Use the English site instead of the original link."""
+        object_num = item.get("object_number")
+        if not object_num:
+            logger.info(
+                f"Image with (foreign) id {item.get('id')} does not have "
+                "`object_number`! Therefore we cannot build the "
+                "foreign_landing_url."
+            )
+            return
+        landing_page_base_url = "https://open.smk.dk/en/artwork/image/"
+        return f"{landing_page_base_url}{object_num}"
+
+    @staticmethod
+    def _get_image_url(image_iiif_id: str, image_size=IMAGE_SIZE):
+        # For high quality IIIF-enabled images, restrict the image size to prevent
+        # loading very large files.
+        # TODO: consider just using the full "image_native" when adding the
+        # "image_thumbnail".
+        image_url = f"{image_iiif_id}/full/!{image_size},/0/default.jpg"
+        return image_url
+
+    @staticmethod
+    def _get_title(item: dict) -> str | None:
+        titles = item.get("titles", [])
+        try:
+            return titles[0].get("title")
+        except (IndexError, KeyError):
+            logger.info(f"No title for image with (foreign) id {item.get('id')}.")
+            return
+
+    @staticmethod
+    def _get_creator(item: dict) -> str | None:
+        # TODO: review this field, there could be more than one creator or artist.
+        # Keeping it as it was for the class refactor.
+        try:
+            return item.get("production", [])[0].get("creator")
+        except (IndexError, KeyError):
+            return
```

Comment:
Same note here RE try/catch

Reply:
Done in [`20fc18d`](https://github.com/WordPress/openverse-catalog/pull/742/commits/20fc18dd56cf44d3862194599db5f98d8883d5b8).